solr/core/src/java/org/apache/solr/security/Permission.java

```diff
@@ -52,18 +53,21 @@ static Permission load(Map<?, ?> m) {
     p.role = readValueAsSet(m, "role");
     if (PermissionNameProvider.Name.get(name) != null) {
       p.wellknownName = PermissionNameProvider.Name.get(name);
-      HashSet<String> disAllowed = new HashSet<>(knownKeys);
-      disAllowed.remove("role"); // these are the only
-      disAllowed.remove(NAME); // allowed keys for well-known permissions
-      disAllowed.remove("collection"); // allowed keys for well-known permissions
-      disAllowed.remove("index");
-      for (String s : disAllowed) {
+      for (String s : customPermissionAdditionalKeys) {
         if (m.containsKey(s))
           throw new SolrException(
               SolrException.ErrorCode.BAD_REQUEST,
               s + " is not a valid key for the permission : " + name);
       }
+    } else if (customPermissionAdditionalKeys.stream().noneMatch(m.keySet()::contains)) {
```

Comment:
It might be better to iterate over stream of `m.keySet()` if it is generally smaller in size. I don't know the details but to me it looks like `m.size()` is generally small!?

Reply:
I tried the other way around but IntelliJ complained about "suspicious" code, and somehow it did not work either. There are only a few values anyway and this code is run once during parsing of security.json, so leaving as is.

Reply:
Funny. Suspicious. 😜

Reply:
You could also use: `m::containsKey`